SignalUITests/SignalUITests.swift

```diff
@@ -0,0 +1,737 @@
+//
+//  SignalUITests.swift
+//  SignalUITests
+//
+//  Created by Matthew Kotila on 1/13/16.
+//  Copyright © 2016 Open Whisper Systems. All rights reserved.
+//
+
+import XCTest
+
+class SignalUITests: XCTestCase {
+    
+    override func setUp() {
+        super.setUp()
+        continueAfterFailure = false
+        XCUIApplication().launch()
+    }
+    
+    override func tearDown() {
+        super.tearDown()
+    }
+    
+    // requires unverified app
```

Comment:
These are just my thoughts on how to do testing well, I don't have commit
access to this project:

It's important that the test suite be able to run without manual
intervention. What I have done with other ios apps is add something like a
testing category to the app delegate to the testing target (only!) which
exposes useful interfaces for faking data.

E.g. addFakeContact, registerPhone, unRegisterPhone, addMessageThread might
be useful for Signal iOS.

Then you can use these methods in your test setup blocks to get the app in
a useful state for your tests.

On Mon, Mar 7, 2016 at 7:09 PM matthewkotila <notifications@github.com>
wrote:

> In SignalUITests/SignalUITests.swift
> <https://github.com/WhisperSystems/Signal-iOS/pull/1049#discussion_r55311097>
> :
>
> > +
> > +import XCTest
> > +
> > +class SignalUITests: XCTestCase {
> > +
> > +    override func setUp() {
> > +        super.setUp()
> > +        continueAfterFailure = false
> > +        XCUIApplication().launch()
> > +    }
> > +
> > +    override func tearDown() {
> > +        super.tearDown()
> > +    }
> > +
> > +    // requires unverified app
>
> Hi Michael, I'm not extremely versed in UI tests, so when I encountered
> the problem of "application state" (i.e. testing UI pre-verification and
> post-verification), I simply wrote tests for an app that was pre-verified,
> and then verified it manually using the iOS simulator with valid
> information, and then wrote tests for the post-verified app. I'm unsure if
> there's a way to automatically input information (such as contacts,
> verification codes, etc.) once and start a different set of UI tests
> intended for a verified app. Additionally, I'm unsure if there's a way to
> automatically "wipe" an app in order to test pre-verification UI. All very
> interesting questions in the realm of UI testing!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/WhisperSystems/Signal-iOS/pull/1049/files#r55311097>.
>